server/bootstrap/passport/strategies/bnet.js

```diff
@@ -10,29 +11,27 @@ module.exports = function(authBnetConfig){
         user
           .findByUserId(profile.id)
           .then(function(){
-            user.setValues({
-                oauthTokenEncrypted: user.encryptOauthToken(accessToken),
-                oauthType: "bnet"
-            }).updateTimeLatestLogin();
-            if (!user.exists()) {
-              user.setValues({
-                  userId: profile.id
-              });
-            }
-            user
-              .save()
-              .then(function(){
-                console.log('Save: success', user.getValues());
-                return done(null, profile);
-              })
-              .error(function(err){
-                console.log("BnetStrategy (2):", err);
-                return done(null, null);
-              });
+            log.debug("Updating existing user. 'profile':\n", profile);
           })
           .error(function(err){
-            console.log("BnetStrategy (1):", err);
-            return done(null, null);
+            log.debug("Creating new user. 'profile':\n", profile);
+          })
+          .finally(function(){
+            user.setValues({
+                oauthTokenEncrypted: user.encryptOauthToken(accessToken),
+                oauthType: "bnet",
+                userId: profile.id
```

Comment:
@Sikian Okay.
Just to be clear, though. There is no "`id`" field in the table. "`userId`" is specified as [the primary key in the User model](https://github.com/Starbow/website/blob/0242fb0b582b54db1d845cfdb6ff7ee2a90e47b8/server/mvc/models/User.js#L106). This means there is only one "id" type field: `userId`.